### ISSUE TITLE
Outdated documentation to be revised

### ISSUE THREAD

**Issue by Syntaf** [STEllAR-GROUP/hpx#1796]
Below is a list of documentation ambiguities and issues i've ran across while reading through.

List of issues and whether they've been fixed: 
  - [x] Fibonacci Example
  - [x] Hello World Example
  - [x] Interest Calculator Example
  - [ ] Futurization Example

### Fibonacci.html
under the `hpx_main()` code block [here](http://stellar-group.github.io/hpx/docs/html/hpx/tutorial/examples/fibonacci.html) , the description looks outdated. Code segment:
```
int hpx_main(boost::program_options::variables_map& vm)
{
    // extract command line argument, i.e. fib(N)
    boost::uint64_t n = vm["n-value"].as<boost::uint64_t>();

    {
        // Keep track of the time required to execute.
        hpx::util::high_resolution_timer t;

        // Wait for fib() to return the value
        fibonacci_action fib;
        boost::uint64_t r = fib(hpx::find_here(), n);

        char const* fmt = "fibonacci(%1%) == %2%\nelapsed time: %3% [s]\n";
        std::cout << (boost::format(fmt) % n % r % t.elapsed());
    }

    return hpx::finalize(); // Handles HPX shutdown
}
```

Documentation explanation:
> Upon a closer look we see that the hpx::future<> we created is assigned the return of 
hpx::async<fibonacci_action>(hpx::find_here(), n). hpx::async<>(), and asynchronously kicks of the computation of the action (fibonacci_action), returning a future which represents the result of the computation. But wait, what is an action? And what is this fibonacci_action? For starters, an action is a wrapper for a function. By wrapping functions, HPX can send packets of work to different processing units. These vehicles allow users to calculate work now, later, or on certain nodes. The first argument to hpx::async<>() is the location where the action should be run. In this case, we just want to run the action on the machine that we are currently on, so we use hpx::find_here(). To further understand this we turn to the code to find where fibonacci_action was defined:

Looks like `fibonacci.cpp` was recently refactored, but the docs weren't updated to reflect these new changes. `r` is no longer of type `hpx::future<>` and the right hand side only calls the action fib, there are no longer async calls. Can someone explain what this new code segment is doing so the docs can be updated properly?

### Hello_world.html
For the explanation of [this](https://github.com/STEllAR-GROUP/hpx/blob/master/examples/quickstart/hello_world.cpp#L110) line in [hello_world.html](http://stellar-group.github.io/hpx/docs/html/hpx/tutorial/examples/hello_world.html), the description of the callback seems to be outdated: 

> hpx::lcos::wait_each() invokes a callback function provided by the user, supplying the callback function with the index of the future in the std::vector<> and the result of the future

from my understanding, `hpx::lcos::wait_each()` does not supply the callback with the index of the future, only the result of the future. This might have never been updated when the example was changed from using `wait()`?

### Interest_calculator.html

Given the code segment:

```
double init_principal=vm["principal"].as<double>(); //Initial principal
double init_rate=vm["rate"].as<double>(); //Interest rate
int cp=vm["cp"].as<int>(); //Length of a compound period
int t=vm["time"].as<int>(); //Length of time money is invested

init_rate/=100; //Rate is a % and must be converted
t/=cp; //Determine how many times to iterate interest calculation:
       //How many full compund periods can fit in the time invested

shared_future<double> principal = make_ready_future(init_principal);
shared_future<double> rate = make_ready_future(init_rate);

for (int i = 0; i < t; ++i)
{
    shared_future<double> interest = dataflow(unwrapped(calc), principal, rate);
    principal = dataflow(unwrapped(add), principal, interest);
}

// wait for the dataflow execution graph to be finished calculating our
// overall interest
double result = principal.get();

std::cout << "Final amount: " << result << std::endl;
std::cout << "Amount made: " << result-init_principal << std::endl;
```

The description in the documentation reads:

> Here we find our command line variables read in, the rate is converted from a percent to a decimal, the number of calculation iterations is determined, and then our dataflows are set up. Notice that we first place our principal and rate into a dataflow by passing the variables p and i_rate to an action called identity_action:

There is no `identity_action`, or `p`, or `i_rate` present in this example. Going back in history it looks likeFuturization

* `p` is now `init_principal`
* `i_rate` is now `init_rate`
*  The example uses shared futures now, there is no mention of this

### Futurization Example

In example 2, under the first code section the paragraph reads

> In example 2, we re-define our partition type as a shared_future and, in main, create the object "result" which is a future to a vector of partitions. We use result to represent the last vector in a string of vectors created for each timestep. In order to move to the next timestep, the values of a partition and its neighbors must be passed to heat once the futures that contain them are ready. In HPX, we have an LCO (Local Control Object) named Dataflow which assists the programmer in expressing this dependency. Dataflow allows us to pass the results of a set of futures to a specified function when the futures are ready. Dataflow takes three types of arguments, one which instructs the dataflow on how to perform the function call (async or sync), the function to call (in this case Op), and futures to the arguments that will be passed to the function. When called, dataflow immediately returns a future to the result of the specified function. This allows users to string dataflows together and construct an execution tree.

there are no objects named `result` inside of the example

**Comment by AntonBikineev** [comment]
Examples section is often the first place where people go. It would be nice to get it updated.

**Comment by Syntaf** [comment]
Issue was resolved by #1798
